Content.Server/Storage/Components/EntityStorageComponent.cs

```diff
@@ -239,6 +239,10 @@ public virtual bool CanFit(EntityUid entity)
             {
                 if (!storageIsItem)
                     allowedToEat = true;
+                else
+                {
+                    allowedToEat = _entMan.TryGetComponent(Owner, out FoldableComponent? foldable) && !foldable.IsFolded;
+                }
             }
 
             _entMan.EventBus.RaiseLocalEvent(entity, allowedToEat);
```

Comment:
allowedToEat shouldn't be getting raised here so delete that for me.
Also you should just raise a different event on the owner instead of adding a tryget here.

Reply:
Done deal-io